### PR TITLE
Problem: slow VerifyEthSig on large number of messages

### PR DESCRIPTION
**Pull request by mmsqe**
Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: app/ante/sigverify.go [hunk 1]
```diff
@@ -29,16 +30,26 @@ import (
 // Failure in RecheckTx will prevent tx to be included into block, especially when CheckTx succeed, in which case user
 // won't see the error message.
 func VerifyEthSig(tx sdk.Tx, signer ethtypes.Signer) error {
+	errChan := make(chan error, len(tx.GetMsgs()))
 	for _, msg := range tx.GetMsgs() {
```
Comment (mmsqe) [inline app/ante/sigverify.go:34]: @yihuang could reduce from 2s to 0.2s for 3k msgs when parallel

Comment (yihuang) [reply app/ante/sigverify.go:34]: Only for large batch tx?

Comment (yihuang) [reply app/ante/sigverify.go:34]: Can you do this conditionally, so we can avoid extra overhead for normal cases where tx has single msg.

Comment (mmsqe) [reply app/ante/sigverify.go:34]: should be same for small batch, but I only filter out when CheckTx > [1s](https://github.com/mmsqe/cosmos-sdk/commit/5472e9cf6643340c1f915eb37369583f3111a634#diff-389d02f089ab55575400f5300d3c5cfe277fde5f04bd0fe6a85456a7252d1099L357)

Comment (yihuang) [reply app/ante/sigverify.go:34]: I mean at least for tx that has a single msg, this change is pure overhead, no benifit.

File: app/ante/sigverify.go [hunk 2]
```diff
@@ -29,16 +30,26 @@ import (
 // Failure in RecheckTx will prevent tx to be included into block, especially when CheckTx succeed, in which case user
 // won't see the error message.
 func VerifyEthSig(tx sdk.Tx, signer ethtypes.Signer) error {
+	errChan := make(chan error, len(tx.GetMsgs()))
```
Comment (yihuang) [inline app/ante/sigverify.go:33]: use [errgroup](https://pkg.go.dev/golang.org/x/sync/errgroup)?

**Pull request closed without merging**